schemas/stsci.edu/asdf/0.1.0/core/column.yaml

```diff
@@ -0,0 +1,47 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/asdf/0.1.0/core/column"
+tag: "tag:stsci.edu:asdf/0.1.0/core/column"
+
+title: >
+  A column in a table.
+
+description: |
+  Each column contains a name and an array of data, and an optional description
+  and unit.
+
+type: object
+properties:
+  name:
+    description: The name of the column
+    type: string
+    pattern: "[A-Za-z_][A-Za-z0-9_]*"
```

Comment:
That's exactly the reason -- to make sure that the column names can be used as token identifiers in any reasonable programming language (note that it's even more restrictive than what Python 3 would support for example).  I think it's an important distinction -- one of the main flaws of VOTable IMHO is that it doesn't do this, and you have to jump through all kinds of messy hoops to make the columns addressable on the Python side.  I wouldn't be opposed to having both this name and some sort of "display name" which would be unrestricted, though, if it's necessary (I can particularly think of the usefulness of Greek math symbols, for example).